rfcs/012-merge-rebase-strategy.md

```diff
@@ -0,0 +1,81 @@
+# RFC 012 - Merge Settings/Merge Strategies
+
+## Affected projects
+
+All the projects under the [Open Collective organization](https://github.com/opencollective).
+
+## Motivation
+
+It's not always clear what merge strategy to use when merging PRs. Currently, core contributors are free to use
+whatever they prefer when merging PRs. GitHub allows three ways,
+
+1) Create a merge commit - All commits from the branch will be added  to the base branch via a merge commit.
+
+2) Squash and merge - All commits will be combined into one commit and added to the base branch.
+
+3) Rebase and merge - All the commits will be rebased and added to the base branch.
+
+### Benefits
+
+- Core contributors can choose any particular method that they see fit. Sometimes a PR would contain unnecessary 
+commits which will pollute the version history and thus a rebase or a squash makes sense. Other times it maybe that
+  all the commits in the PR adds value. 
+  
+### Downsides
+
+The main downside of this is that we are assuming that the person who made the branch/PR will keep the Git history clean
+on that branch. For this purpose we have guidelines; https://github.com/opencollective/opencollective-api/blob/main/CONTRIBUTING.md. 
+
+These guidelines propose interactive rebasing which comes with the following downsides.
+
+1) Most new contributors and especially ones who are not familiar with Git will find this confusing.
+
+2) Doing interactive re-bases for larger PRs can be time consuming when there are conflicts. Since the conflicts
+needs to be solved for each commit separately rather than as a whole. While merging seems easier since it takes the whole diff
+   and applies it in one go.
```

Comment:
I don't see the advantage in term of conflict solving.

Before Squasing and Merge, we still need to resolve conflict and there is no difference with a merge commit strategy.

Reply:
Edit: Ok, I get it, the point is to let people merge from main if they want and stop caring.

Reply:
Yes, my thinking is some people might prefer the merge; especially for people who aren't used much to Git I think this is what they would like to do. In any case I think we should try to limit people especially external contributors from doing merges in any way they want. The three projects also seems to allow for merges. 😄 🐨

Reply:
If you like me to add any details to this to make it clearer let me know as well. 😄